app/src/main/java/fr/neamar/kiss/db/DBHelper.java

```diff
@@ -231,6 +231,7 @@ public static boolean insertShortcut(Context context, ShortcutRecord shortcut) {
         Cursor cursor = db.query("shortcuts", new String[]{"package", "intent_uri"},
                 "package = ? AND intent_uri = ?", new String[]{shortcut.packageName, shortcut.intentUri}, null, null, null, null);
         if (cursor.moveToFirst()) {
+            cursor.close();
```

Comment:
Please wrap code in try-with-resources statement here instead of calling ```cursor.close()``` per branch.
This will ensure that cursor is closed, even when exceptions are thrown from within the try block.